Added script to edit copied PIConGPU checkpoints.

nicowrobel opened the pull request:
The python script can be used to copy checkpoints and add particles to the copy. 
The added Jupyter Notebook shows, how this could be used to add a driver into an empty checkpoint.
A lot of code from the [pipe script](https://github.com/openPMD/openPMD-api/blob/dev/src/binding/python/openpmd_api/pipe/__main__.py) from the openPMD-api was used, therfore Franz Poeschel is credited.
Tests for the current dev and, if needed, hdf5 checkpoints are pending.

Supported data for particle species inputs are arrays with the position, momentum and weighting of the particles.
probeE, probeB and id data is supported for the new particles.
Currently only overwriting checkpoint 0 is supported and only if there aren't already particles of the given species.

nicowrobel committed:
Added script to edit copied PIConGPU checkpoints.

The python script can be used to copy checkpoints and add particles
to the copy. The added Jupyter Notebook shows, how this could be
used to add a driver into an empty checkpoint.

nicowrobel committed:
Removed whitespaces

According to the pep8 standard.

nicowrobel committed:
Fixes to be pep8 conform

Removed whitespaces, followed the character limit

nicowrobel committed:
Fixed most formating errors

Found out, that i can see an extensive log of all errors and tried to fix them.
Some "whitespace before [" errors remain and will be fixed.

nicowrobel committed:
Fixed all style errors

This should make it complient to pep8.

PrometheusPi commented on lib/python/picongpu/extra/input/createBunch_example.ipynb:
```diff
@@ -0,0 +1,294 @@
+    "This notebook allows you to define a PWFA driver bunch using numpy.\n",
+    "Each step will be explained on the way.\n",
+    "In the end, you will add your particles (of the driver bunch) to an empty **openPMD-api** checkpoint in **HDF5** format.\n",
+    "Since rewriting the entire code would have taken too long (before you start Nico), you have to stick to **HDF5** for now. "
```
"before you start Nico" 😉

PrometheusPi commented on lib/python/picongpu/extra/input/createBunch_example.ipynb:
```diff
@@ -0,0 +1,294 @@
+   "source": [
+    "## convert data and write to checkpoint"
+   ]
+  },
```
This looks like a heading to a removed piece of code, is this right?

PrometheusPi commented on lib/python/picongpu/extra/input/createBunch_example.ipynb:
```diff
@@ -0,0 +1,294 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# assigne to (existing) checkpoint file (needs to be hdf5 file)\n",
```
"needs to be hdf5 file" - no longer true

PrometheusPi reviewed (dismissed):
looks good to me - just some minor questions/comments, perhaps we can clarify this before merging

nicowrobel committed:
Fixed error, added better instructions to notebook

This commmit fixes an breaking bug, caused by missing brickets
around a multiline command. Additional comments were added to the
notebook. Unused variable assignements and modules were removed and
left-over comments removed.
